src/App.tsx

```diff
@@ -101,16 +103,18 @@ const App: React.FC<AppProps> = ({plugins}) => {
       return;
     }
 
-    getApiDetails(apiEndpoint).catch(() => {
-      // Handle race condition
-      if (getApiEndpoint() !== apiEndpoint) {
-        return;
-      }
-
-      // Display popup
-      notificationCall('failed', 'Could not receive data from the specified API endpoint');
-      setEndpointModalState(true);
-    });
+    getApiDetails(apiEndpoint)
```

Comment:
This effect is called only in case the `REACT_APP_API_SERVER_ENDPOINT` is not set. When it will be set, it won't run:

https://testkube-dashboard-git-pavlo-display-version-testkube-cloud.vercel.app/settings?~api_server_endpoint=https://dev.testkube.io/results/v1

Reply:
* The best solution would be to replace this `useEffect` with `useAsync` and `useEffect`,
* The easiest way would be to move `isApiEndpointLocked()` condition to the condition below:
  * `location.pathname === '/apiEndpoint'` instead of `location.pathname === '/apiEndpoint' || isApiEndpointLocked()`
  * `!apiEndpoint && !isApiEndpointLocked()` instead of `!apiEndpoint`

Reply:
done